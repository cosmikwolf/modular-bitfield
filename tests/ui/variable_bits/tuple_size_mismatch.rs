use modular_bitfield::prelude::*;

#[derive(Specifier)]
#[variable_bits(8, 16)]  // Only 2 sizes but 3 variants
enum TestData {
    #[discriminant = 0]
    Small(u8),
    #[discriminant = 1]
    Medium(u16),
    #[discriminant = 2]
    Large(u32),
}

fn main() {}