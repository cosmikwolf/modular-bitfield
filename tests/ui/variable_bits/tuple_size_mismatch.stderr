error: variable_bits tuple must have 3 elements (one per variant), found 2
  --> tests/ui/variable_bits/tuple_size_mismatch.rs:4:1
   |
4  | / #[bits(8, 16)]  // Only 2 sizes but 3 variants
5  | | enum TestData {
6  | |     #[discriminant = 0]
7  | |     Small(u8),
...  |
11 | |     Large(u32),
12 | | }
   | |_^
